xbmc/cores/VideoPlayer/test/edl/TestEdl.cpp

```diff
@@ -0,0 +1,421 @@
+  // fps was not supplied, kodi will not be able to process the file
+  EXPECT_EQ(found, false);
+  // parse the file again this time supplying 60 fps
+  const float fps = 60;
+  found = edl.ReadEditDecisionLists(mediaItem, fps);
+  EXPECT_EQ(found, true);
+  EXPECT_EQ(edl.HasEdits(), true);
+  EXPECT_EQ(edl.GetCutMarkers().empty(), true);
+  EXPECT_EQ(edl.GetEditList().size(), 3);
+  EXPECT_EQ(edl.GetSceneMarkers().size(), 3 * 2); // start and end of each commbreak
+  EXPECT_EQ(edl.GetEditList().front().start, static_cast<int>(12693 / fps * 1000));
+  EXPECT_EQ(edl.GetEditList().front().end, static_cast<int>(17792 / fps * 1000));
+}
+
+TEST_F(TestEdl, TestComSkipVersion2EDL)
+{
+  CEdl edl;
+
+  // create a dummy "media" fileitem whose corresponding edl file is testdata/comskipversion2.txt
+  // this is an edl file in ComSkip (version 2) format where fps is obtained from the file
+  CFileItem mediaItem;
+  mediaItem.SetPath(
+      XBMC_REF_FILE_PATH("xbmc/cores/VideoPlayer/test/edl/testdata/comskipversion2.mkv"));
+  const bool found = edl.ReadEditDecisionLists(mediaItem, 0);
+  EXPECT_EQ(found, true);
+  // fps is obtained from the file as it always takes precedence (note we supplied 0 above),
+  // the EDL file has the value of 2500 for fps. kodi converts this to 25 fps by dividing by a factor of 100
+  const float fpsInEdlFile = 2500 / 100;
+  // this format only supports commbreak types
+  EXPECT_EQ(edl.HasEdits(), true);
+  EXPECT_EQ(edl.GetCutMarkers().empty(), true);
+  EXPECT_EQ(edl.GetEditList().size(), 3);
+  EXPECT_EQ(edl.GetSceneMarkers().size(), 3 * 2); // start and end of each commbreak
+  EXPECT_EQ(edl.GetEditList().front().start, static_cast<int>(12693 / fpsInEdlFile * 1000));
+  EXPECT_EQ(edl.GetEditList().front().end, static_cast<int>(17792 / fpsInEdlFile * 1000));
+}
+
+TEST_F(TestEdl, TestRuntimeSetEDL)
+{
+  // this is a simple test for SetLastEditTime, SetLastEditActionType and corresponding getters
+  CEdl edl;
+  edl.SetLastEditTime(1000);
+  edl.SetLastEditActionType(Action::COMM_BREAK);
+  EXPECT_EQ(edl.GetLastEditTime(), 1000);
+  EXPECT_EQ(edl.GetLastEditActionType(), Action::COMM_BREAK);
+}
+
+TEST_F(TestEdl, TestCommBreakAdvancedSettings)
+{
+  CEdl edl;
+  const std::shared_ptr<CAdvancedSettings> advancedSettings =
+      CServiceBroker::GetSettingsComponent()->GetAdvancedSettings();
+  // make sure autowind and autowait are set to default values 0
+  EXPECT_EQ(advancedSettings->m_iEdlCommBreakAutowait, 0);
+  EXPECT_EQ(advancedSettings->m_iEdlCommBreakAutowind, 0);
+  // create a dummy "media" fileitem whose corresponding edl file is testdata/edlautowindautowait.txt
+  CFileItem mediaItem;
+  mediaItem.SetPath(
+      XBMC_REF_FILE_PATH("xbmc/cores/VideoPlayer/test/edl/testdata/edlautowindautowait.mkv"));
+  bool found = edl.ReadEditDecisionLists(mediaItem, 0);
+  EXPECT_EQ(found, true);
+  // confirm the start and end times of all the commbreaks match
+  EXPECT_EQ(edl.GetEditList().size(), 5);
```

Comment:
I really have doubts here. The EDL file has 6 commbreaks defined in the file:

```
## first commbreak should be automatically removed
0    1     3
10   22    3
30   32    3
37   50    3
52   60    3
62   65.1  3
```

However kodi removes the first one automatically because of this:
https://github.com/xbmc/xbmc/blob/30f7803d80be535479c755303bf1a8443d8e8dcb/xbmc/cores/VideoPlayer/Edl.cpp#L936-L949

This occurs in the scope of `MergeShortCommbreaks`, even though even if the mergecommbreaks advanced setting is false it will run anyway. So there is no way to avoid this behaviour nor to specify what "start" means. Kodi just takes the first commercial break and removes it regardless of the position. I think we should have a way (advanced settings) to:
- Disable this behaviour
- Define what start means